### DISCUSSION TITLE
ComPtr how to detect it is Empty or null  or not init ?

### DISCUSSION THREAD

**Discussion by sgf** [dotnet/Silk.NET#1821] [Q&A]
```C#
        /// <summary>
        /// Initializes a new SolidBrush for the given Graphics device using a transparent Color.
        /// </summary>
        /// <param name="renderTarget">A Graphics device.</param>
        public SolidBrush(ComPtr<ID2D1HwndRenderTarget> renderTarget):this()
        {
            if (renderTarget!=) throw new ArgumentNullException(nameof(renderTarget));
            renderTarget.CreateSolidColorBrush(default, new BrushProperties(), ref _brush);
        }
```

` if (renderTarget!=)`

At the same time, I am not sure which variable definitions in Silk.NET need to be surrounded by ComPtr<> and which do not. I am not good at C++ and I am not good at using smart pointers.
So I don't know anything about it.
If I define ComPtr<?> as a parameter, can I pass it directly without applying ref and other description (like  ComPtr<?>* or ComPtr<?>** or ref ComPtr<?>*  etc..   )?

**Comment by Beyley** [comment]
You can just check if its `default` or not, since `default` will be a nullptr

**Reply by sgf** [reply]
I've tried it but it doesn't support it.
![image](https://github.com/dotnet/Silk.NET/assets/2557696/2d7e0ad4-38b3-4e9b-bc04-445df20860aa)

**Reply by qian-o** [reply]
The ComPtr has a Handle member, which can be used to determine.

**Comment by sgf** [comment]
I have switched to DirectN now. Although DirectN also has some problems.
But Silk seems to have more problems.
Typical example: Matrix3x2<float> cannot be multiplied by Matrix3x2<float>.
Or conversion between ID2DSolidColorBrush and ID2DBrush (operator overloading only supports one-way conversion).

There are many more questions like this, so I won’t list them all.
Thanks for your patience in answering.


Silk's packaging is very clever, but it doesn't look perfect yet. When I encounter some problems, it is difficult to solve them in simple ways. Or because ease of use and performance are inherently incompatible.
